src/array/data_chunk.rs

```diff
@@ -129,6 +129,23 @@ impl DataChunk {
         }
         arrays.into_iter().collect()
     }
+
+    pub fn stack(&mut self, chunks: Vec<data_chunk::DataChunk>) {
+        let mut arrays = vec![];
+        for col_idx in 0..self.column_count() {
+            let mut builder = ArrayBuilderImpl::from_type_of_array(self.array_at(col_idx));
+            for row in self.rows() {
+                builder.push(&row.get(col_idx))
+            }
+            for chunk in &chunks {
+                for row in chunk.rows() {
+                    builder.push(&row.get(col_idx))
+                }
+            }
```

Comment:
`ArrayBuilder` can `append` an array directly. We don't need to `push` row by row